agent/engine/docker_task_engine.go

```diff
@@ -625,7 +654,19 @@ func (engine *DockerTaskEngine) startContainer(task *api.Task, container *api.Co
 			Error: CannotStartContainerError{fmt.Errorf("Container not recorded as created")},
 		}
 	}
-	return client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+	dockerContainerMD := client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+
+	// Get metadata through container inspection and available task information then write this to the metadata file
+	// Performs this in the background to avoid delaying container start
+	if dockerContainerMD.Error == nil && engine.cfg.ContainerMetadataEnabled == true {
```

Comment:
I am not sure what you mean here. client.StartContainer only returns a DockerContainerMetadata which contains an error field and I do check for this error. The reason I do not change this behavior is because this feels beyond the scope of this project and will probably require significant refactoring in the engine package.